Lib/test/test_zipfile.py

```diff
@@ -1442,6 +1442,16 @@ def test_read0(self):
 
                 self.assertEqual(f.read(), b"O, for a Muse of Fire!")
 
+    def test_seek_before_current_position(self):
+        """Check that calling seek() at a position before the current one
+        on a ZipExtFile object resets its state and still works."""
+        with zipfile.ZipFile(TESTFN, mode="w") as zipf:
+            zipf.writestr("foo.txt", "O, for a Muse of Fire!")
+            with zipf.open("foo.txt") as f:
+                f.seek(f._orig_file_size)
+                f.seek(0)
+                self.assertEqual(f.read(), b"O, for a Muse of Fire!")
+
```

Comment:
The test essentially asserts that `f.seek(f._orig_file_size); f.seek(0)` does nothing. It would pass even if `f.seek` did nothing at all.
What do you think about e.g. replacing `f.seek(f._orig_file_size)` by `f.read()`?

Reply:
Good catch, thank you!